src/Lean/PrettyPrinter/Delaborator/Basic.lean

```diff
@@ -240,6 +267,9 @@ partial def delabFor : Name → Delab
     <|> if k.isAtomic then failure else delabFor k.getRoot
 
 partial def delab : Delab := do
+  if ← isMaxDepthReached then
```

Comment:
For atomic expressions, it seems to me that there's no need to replace them with `⋯`. There's a chance an atomic expression has a pretty printer that does something complicated (like print a type ascription), but then that type ascription will likely get a `⋯`.

```suggestion
  if ← pure !e.isAtomic <&&> isMaxDepthReached then
```

In any case, I think for consistency we should move this to after `checkSystem`.

Reply:
I generalized this idea to "shallow expressions": The delaborator now never omits subterms if they have an `approxDepth <= maxTermDepth / 4`.
Without this kind of generalization, `[1, 2, 3, 4, 5]` would print as `[1, 2, 3, 4, ⋯, ⋯]`, with `⋯` referring to `5` and `[]` respectively because both terms are not atomic, but also exceed the `maxTermDepth` at that point.

Please let me know if you want to keep this generalization.